pkg/vulnerability/vulnerability.go

```diff
@@ -40,21 +42,45 @@ func NewClient(dbc db.Config) Client {
 	return Client{dbc: dbc}
 }
 
-func (c Client) FillInfo(vulns []types.DetectedVulnerability, light bool) {
+func (c Client) FillInfo(vulns []types.DetectedVulnerability, reportType string) {
 	var err error
-	var severity dbTypes.Severity
 
 	for i := range vulns {
-		if light {
-			severity, err = c.dbc.GetSeverity(vulns[i].VulnerabilityID)
-			vulns[i].Vulnerability.Severity = severity.String()
-		} else {
-			vulns[i].Vulnerability, err = c.dbc.GetVulnerability(vulns[i].VulnerabilityID)
+		vulns[i].Vulnerability, err = c.dbc.GetVulnerability(vulns[i].VulnerabilityID)
+		switch reportType {
+		case vulnerability.Ubuntu, vulnerability.Alpine, vulnerability.RedHat, vulnerability.RedHatOVAL, vulnerability.Debian, vulnerability.DebianOVAL, vulnerability.CentOS, vulnerability.Fedora, vulnerability.Amazon, vulnerability.OracleOVAL, vulnerability.SuseCVRF, vulnerability.OpenSuseCVRF, vulnerability.Photon:
+			c.getVendorSeverity(&vulns[i], reportType)
+		case "npm", "yarn":
+			c.getVendorSeverity(&vulns[i], vulnerability.NodejsSecurityWg)
+		case "pipenv", "poetry":
+			c.getVendorSeverity(&vulns[i], vulnerability.PythonSafetyDB)
+		case "bundler":
+			c.getVendorSeverity(&vulns[i], vulnerability.RubySec)
+		case "cargo":
+			c.getVendorSeverity(&vulns[i], vulnerability.RustSec)
+		case "composer":
+			c.getVendorSeverity(&vulns[i], vulnerability.PhpSecurityAdvisories)
 		}
```

Comment:
nit:
```suggestion
		var source string
		switch reportType {
		case vulnerability.Ubuntu, vulnerability.Alpine, vulnerability.RedHat, vulnerability.RedHatOVAL, vulnerability.Debian, vulnerability.DebianOVAL, vulnerability.CentOS, vulnerability.Fedora, vulnerability.Amazon, vulnerability.OracleOVAL, vulnerability.SuseCVRF, vulnerability.OpenSuseCVRF, vulnerability.Photon:
			source = reportType
		case "npm", "yarn":
			source = vulnerability.NodejsSecurityWg
		case "pipenv", "poetry":
			source = vulnerability.PythonSafetyDB
		case "bundler":
			source = vulnerability.RubySec
		case "cargo":
			source = vulnerability.RustSec
		case "composer":
			source = vulnerability.PhpSecurityAdvisories
		c.getVendorSeverity(&vulns[i], source)
```
```

Reply:
I think adding the temporal variable of source doesn't really help with duplication. I'd like to leave it as is unless I missed a point.

Reply:
OK. But I want to know why it doesn't help. If the signature of getVendorSeverity is changed, we have to fix all of them.

Reply:
Fair enough, I refactored it.